litestar/plugins/problem_details.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, Callable, Mapping, TypeVar
+
+from litestar.connection.request import Request
+from litestar.exceptions.http_exceptions import HTTPException
+from litestar.plugins.base import InitPluginProtocol
+from litestar.response.base import Response
+from litestar.serialization.msgspec_hooks import encode_json
+from litestar.types.callable_types import ExceptionHandler, ExceptionT
+
+if TYPE_CHECKING:
+    from litestar.config.app import AppConfig
+
+ProblemDetailsExceptionT = TypeVar("ProblemDetailsExceptionT", bound="ProblemDetailsException")
+ProblemDetailsExceptionHandlerType = Callable[[Request, ProblemDetailsExceptionT], Response]
+ExceptionToProblemDetailMapType = Mapping[type[ExceptionT], Callable[[ExceptionT], ProblemDetailsExceptionT]]
+
+
+def _problem_details_exception_handler(request: Request, exc: ProblemDetailsException) -> Response:
+    return exc.to_response(request)
+
+
+def _create_exception_handler(
+    exc_to_problem_details_exc_fn: Callable[[ExceptionT], ProblemDetailsException], exc_type: type[ExceptionT]
+) -> ExceptionHandler:
+    def _exception_handler(req: Request, exc: exc_type) -> Response:
+        problem_details_exc = exc_to_problem_details_exc_fn(exc)
+
+        return problem_details_exc.to_response(req)
+
+    return _exception_handler
+
+
+class ProblemDetailsException(HTTPException):
+    _PROBLEM_DETAIL_HEADER = "application/problem+json"
+
+    def __init__(
```

Comment:
I don't see a benefit in making these parameters `| None` here:

```python
title: str = '',
instance: str = '',
detail: str | dict[str, Any] | list[Any] = '',
```
Also notice that you changed the default value of `detail` from `''` to `None`. I don't see any benefit to it. Is there any?

The other parameters are best to follow what happens in super-class though so totally understand and would like to see them `| None`.

Reply:
Aah I didn't really put that much thought into this when I wrote this, but I agree it should be similar to the super class.